src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php

```diff
@@ -178,18 +194,17 @@ protected function handleCircularReference($object)
     /**
      * Format an attribute name, for example to convert a snake_case name to camelCase.
      *
+     * @deprecated since 2.7, to be removed in 3.0. Use Symfony\Component\Serializer\NameConverter\CamelCaseToSnakeCaseNameConverter instead.
+     *
      * @param string $attributeName
+     *
      * @return string
      */
     protected function formatAttribute($attributeName)
     {
-        if (in_array($attributeName, $this->camelizedAttributes)) {
-            return preg_replace_callback('/(^|_|\.)+(.)/', function ($match) {
-                return ('.' === $match[1] ? '_' : '').strtoupper($match[2]);
-            }, $attributeName);
-        }
+        trigger_error('The formatAttribute() method has been deprecated. Use Symfony\Component\Serializer\NameConverter\CamelCaseToSnakeCaseNameConverter instead.', E_USER_DEPRECATED);
```

Comment:
Using `__METHOD__` would be better than using `formatAttribute()` (otherwise you don't know the method's class).